src/mrack/providers/aws.py

```diff
@@ -209,8 +212,87 @@ async def validate_hosts(self, reqs):
         """Validate that host requirements are well specified."""
         return
 
-    async def can_provision(self, hosts):
-        """Check that hosts can be provisioned."""
+    async def get_subnet_available_ips(self, subnet_id, log_msg_start):
+        """Get number of IPs available in a subnet."""
+        try:
+            subnet = self.ec2.Subnet(subnet_id)
+        except ClientError:
+            logger.warning(
+                f"{log_msg_start} Error retrieving info from subnet: {subnet_id}"
+            )
+            return 0
+
+        logger.debug(
+            f"{log_msg_start} Subnet Id: {subnet_id}\n"
+            f"\tHas {subnet.available_ip_address_count} IP addresses available"
+        )
```

Comment:
This is a nitpick but It would be nice to cahgne output to something like (without even `log_msg_start`):
```suggestion
        logger.debug(f"Subnet Id: {subnet_id}")
        logger.debug(f"  available: {subnet.available_ip_address_count}")
```

To look similar like:
```
   2022-10-27 01:35:04,175 mrack.providers.openstack DEBUG Network: foo_network
    2022-10-27 01:35:04,175 mrack.providers.openstack DEBUG   total: 251
    2022-10-27 01:35:04,175 mrack.providers.openstack DEBUG   used: 105
    2022-10-27 01:35:04,175 mrack.providers.openstack DEBUG   available: 146
```

Reply:
I have update it to look like this (I left log_msg_start bc I think it's useful to spot things in the log):
```
    2022-11-02 11:37:21,077 mrack.providers.aws DEBUG AWS Subnet subnet-08bd210025a255be9
    2022-11-02 11:37:21,483 mrack.providers.aws DEBUG AWS   available: 58
    2022-11-02 11:37:21,484 mrack.providers.aws DEBUG AWS Subnet subnet-0119dfca46c7527f4
    2022-11-02 11:37:21,720 mrack.providers.aws DEBUG AWS   available: 251
```